src/components/Login/Store.tsx

```diff
@@ -0,0 +1,63 @@
+import { EventEmitter } from "events";
+import { IAction } from "../../Models/ActionCreation";
+import * as Actions from "./Actions"
+import dispatcher from "../dispatcher"
+import { Status } from "../PlayerStats/models/Status";
+import {IUser} from "../../Models/DatabaseModels";
+
+class LoginStore extends EventEmitter {
+
+    isLoggedIn: boolean;
+    requestStatus: Status;
+    connectedUser: IUser;
+    error: string;
+    token: string;
+
+    constructor() {
+        super();
+        this.isLoggedIn = false;
+        this.requestStatus = Status.Idle;
+    }
+
+    getRequestStatus(): Status {
+        return this.requestStatus;
+    }
+
+    userIsLoggedIn():boolean{
+        return this.isLoggedIn == true;
+    }
+
+    isAdmin(): boolean {
+        return this.isLoggedIn && this.connectedUser.is_admin;
+    }
+
+    getConnectedUser(): IUser {
+        return this.connectedUser;
+    }
+
+    handleActions(action: IAction) {
+        switch (action.type) {
+            case "AUTHENTICATE_USER":
+                this.requestStatus = Status.Started;
+                this.emit("requestState")
```

Comment:
Pour ma part, la string retournée par this.emit() est toujours la même que le action.type. Ca nous évite d'avoir à deviner de quelle action ça découle. Pourrais tu faire ca aussi?